duckduckgo/macos-browser
# Add pixels related to Duck Player usage

Task/Issue URL: https://app.asana.com/0/1177771139624306/1205296919343346/f

**Description**:
This change adds several pixels to learn about Duck Player usage patterns.

**Steps to test this PR**:
1. Run the app from Xcode
2. Enable Pixel Logging from Main Menu -> Debug
3. Go to Settings -> Duck Player
4. Select "Always open YouTube videos Duck Player", verify that `m.mac.duck-player.setting.always` is fired
5. Select "Never use Duck Player", verify that `m.mac.duck-player.setting.never` is fired
6. Select the default option, verify that `m.mac.duck-player.setting.back-to-default` is fired
7. Go to YouTube and to a video page.
8. Click "Watch here" and verify that no pixels are fired
9. Refresh the page, click "Watch in Duck Player" and verify that `m.mac.duck-player.view-from.youtube.main-overlay` was fired and also that `m.mac.duck-player.daily-unique-view` was fired
10. Refresh the page, check "Remember my choice", click "Watch in Duck Player" and verify that `m.mac.duck-player.view-from.youtube.main-overlay` was fired, `m.mac.duck-player.setting.always` was fired, and also that `m.mac.duck-player.daily-unique-view` was not fired
11. Search for something on SERP
12. Open videos vertical
13. Click on any video, verify that `m.mac.duck-player.view-from.serp` was fired
14. Go to Settings -> Duck Player, switch back to the default setting.
15. Refresh videos vertical, click on any video, click "Watch in Duck Player" and verify that `m.mac.duck-player.view-from.serp` was fired
16. Go back to videos verticall, click on any video, check "Remember my choice" and click "Watch in Duck Player", verify that `m.mac.duck-player.setting.always` and `m.mac.duck-player.view-from.serp` were fired
17. Go to YouTube and click on a video; verify that `m.mac.duck-player.view-from.youtube.automatic` was fired.
18. Open YouTube video directly: https://www.youtube.com/watch?v=dQw4w9WgXcQ; verify that `m.mac.duck-player.view-from.other` was fired.
19. Repeat the previous step with the following URL: https://youtu.be/dQw4w9WgXcQ; verify that `m.mac.duck-player.view-from.other` was fired.
20. Disable Duck Player, open YouTube URL and verify that no pixels are fired.
21. Set Duck Player to the default setting, go to youtube.com, click on a dax icon on a thumbnail overlay. Verify that `m.mac.duck-player.view-from.youtube.hover-button` was fired.



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
